pkg/ostree/ostree.go

```diff
@@ -234,8 +240,30 @@ func Resolve(source SourceSpec) (CommitSpec, error) {
 		URL: source.URL,
 	}
 
+	if source.RHSM && source.MTLS != nil {
+		return commit, NewResolveRefError("cannot use both RHSM and MTLS when resolving ref")
+	}
+
 	if source.RHSM {
+		var subs *rhsm.Subscriptions
+		var err error
+
 		commit.Secrets = "org.osbuild.rhsm.consumer"
+		subs, err = rhsm.LoadSystemSubscriptions()
+
+		if err != nil {
+			return commit, NewResolveRefError("error adding rhsm certificates when resolving ref: %s", err)
+		}
+
+		if subs.Consumer == nil {
+			return commit, NewResolveRefError("error adding rhsm certificates when resolving ref")
+		}
+
+		source.MTLS = &MTLS{}
+		source.MTLS.ClientCert = subs.Consumer.ConsumerCert
+		source.MTLS.ClientKey = subs.Consumer.ConsumerKey
```

Comment:
This would be more readable if the struct was assigned directly:
```go
source.MTLS = &MTLS{
  ClientCert: subs.Consumer.ConsumerCert
  ClientKey: subs.Consumer.ConsumerKey
}
```